layers/core_checks/cc_device.cpp

```diff
@@ -942,3 +956,73 @@ bool CoreChecks::PreCallValidateGetDeviceFaultInfoEXT(VkDevice device, VkDeviceF
     }
     return skip;
 }
+
+bool CoreChecks::PreCallValidateCreatePipelineBinariesKHR(VkDevice device, const VkPipelineBinaryCreateInfoKHR *pCreateInfo,
+                                                          const VkAllocationCallbacks *pAllocator,
+                                                          VkPipelineBinaryHandlesInfoKHR *pBinaries,
+                                                          const ErrorObject &error_obj) const {
+
+    bool skip = false;
+
+    uint32_t pointerCount = 0;
+
+
+    if (pCreateInfo->pipeline != VK_NULL_HANDLE) {
+        auto pipeline = pCreateInfo->pipeline;
+        auto pipeline_state = Get<vvl::Pipeline>(pipeline);
+
+        ASSERT_AND_RETURN_SKIP(pipeline_state);
+
+        if (!(pipeline_state->create_flags & VK_PIPELINE_CREATE_2_CAPTURE_DATA_BIT_KHR)) {
+            skip |= LogError("VUID-VkPipelineBinaryCreateInfoKHR-pipeline-09607", pipeline,
+                             error_obj.location.dot(Field::pCreateInfo).dot(Field::pipeline),
+                             "called on a pipeline created without the "
+                             "VK_PIPELINE_CREATE_2_CAPTURE_DATA_BIT_KHR flag set.");
+        }
+
+        if (pipeline_state->binary_data_released) {
+            skip |= LogError("VUID-VkPipelineBinaryCreateInfoKHR-pipeline-09608", pipeline,
+                             error_obj.location.dot(Field::pCreateInfo).dot(Field::pipeline),
+                             "called on a pipeline after vkReleaseCapturedPipelineDataKHR was called on it.");
+        }
+
+        pointerCount++;
+    }
+
+    if (pCreateInfo->pPipelineCreateInfo != nullptr) {
+        auto *props = &phys_dev_ext_props.pipeline_binary_props;
+
+        if (!props->pipelineBinaryInternalCache) {
+            skip |= LogError("VUID-VkPipelineBinaryCreateInfoKHR-pipelineBinaryInternalCache-09609", device,
+                             error_obj.location.dot(Field::pCreateInfo).dot(Field::pPipelineCreateInfo),
+                             "called when pipelineBinaryInternalCache is false, and pPipelineCreateInfo not nullptr.");
```

Comment:
`error_obj.location.dot(Field::pCreateInfo).dot(Field::pPipelineCreateInfo),` will print out the location, so the error message should be in that order

```suggestion
                             "is not NULL, but pipelineBinaryInternalCache is false.");
```